lib/web_ui/lib/src/engine/shader_data.dart

```diff
@@ -47,14 +52,30 @@ class ShaderData {
       if (type == UniformType.SampledImage) {
         textureCount += 1;
       } else {
-        final Object? rows = rawUniformData['rows'];
         final Object? bitWidth = rawUniformData['bit_width'];
-        if (bitWidth is! int || rows is! int) {
+
+        final Object? arrayElements = rawUniformData['array_elements'];
+        final Object? rows = rawUniformData['rows'];
+        final Object? columns = rawUniformData['columns'];
+
+        if (bitWidth is! int ||
+            rows is! int ||
+            arrayElements is! int ||
+            columns is! int) {
           throw const FormatException('Invalid Shader Data');
         }
-        floatCount += (bitWidth ~/ 32) * rows;
+
+        final int units = rows * columns;
+
+        int value = (bitWidth ~/ 32) * units;
+
+        if (arrayElements > 1) {
+          value *= arrayElements;
+        }
+
+        floatCount += value;
       }
-      uniforms[location] = UniformData(
+      uniforms[i] = UniformData(
```

Comment:
To reviewer: Attention to this change, in the current impl we get an out of bounds error as the location value also considers the array size.